src/datepicker/types.js

```diff
@@ -53,109 +58,116 @@ export type DatepickerOverridesT = {
   Popover?: OverrideT,
 };
 
-export type DayPropsT = {
+export type DayPropsT<T = Date> = {
```

Comment:
Which there are a lot of changes to these types, almost all of them(aside from the addition of the adapter prop) should just be switching from `Date` to `T`